### ISSUE TITLE
langchain.schema.OutputParserException: Could not parse LLM output: `Thought: Do I need to use a tool? No

### ISSUE THREAD

**Issue by pradeepdev-1995** [langchain-ai/langchain#7480]
### System Info

langchain==0.0.219
python 3.9

### Information

- [ ] The official example notebooks/scripts
- [ ] My own modified scripts

### Related Components

- [X] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

```
import os
from llama_index import LLMPredictor,ServiceContext,LangchainEmbedding
from langchain.embeddings.huggingface import HuggingFaceEmbeddings
from langchain.agents import Tool
from langchain.chains.conversation.memory import ConversationBufferMemory
from langchain.chat_models import AzureChatOpenAI

BASE_URL = "url"
API_KEY = "key"
DEPLOYMENT_NAME = "deployment_name"

model = AzureChatOpenAI(
    openai_api_base=BASE_URL,
    openai_api_version="version",
    deployment_name=DEPLOYMENT_NAME,
    openai_api_key=API_KEY,
    openai_api_type="azure",
)

from langchain.agents import initialize_agent

from llama_index import VectorStoreIndex, SimpleDirectoryReader

documents = SimpleDirectoryReader("/Data").load_data()
llm_predictor = LLMPredictor(llm=model)
embed_model = LangchainEmbedding(HuggingFaceEmbeddings(model_name='huggingface model'))


service_context = ServiceContext.from_defaults(llm_predictor=llm_predictor,embed_model=embed_model)

index = VectorStoreIndex.from_documents(documents=documents,service_context=service_context)

tools = [
    Tool(
        name="LlamaIndex",
        func=lambda q: str(index.as_query_engine().query(q)),
        description="useful for when you want to answer questions about the author. The input to this tool should be a complete english sentence.",
        return_direct=True,
    ),
]

memory = ConversationBufferMemory(memory_key="chat_history")
agent_executor = initialize_agent(
    tools, model, agent="conversational-react-description", memory=memory
)

while True:
    query = input("Enter query\n")
    print(agent_executor.run(input=query))

```
Trying the above code, but when i ask queries, it shows the error - '**langchain.schema.OutputParserException: Could not parse LLM output: `Thought: Do I need to use a tool? No**'

### Expected behavior

The error should not occur

**Comment by kashirin-dm** [comment]
The same!

**Comment by yousenwang** [comment]
#549

**Comment by Ecolow** [comment]
I think I found a pretty good fix: 

```
class ConversationalAgent(Agent):
    """An agent that holds a conversation in addition to using tools."""

   #  ...   we don't care  ....

    @property
    def llm_prefix(self) -> str:
        """Prefix to append the llm call with."""
        return "New Thought Chain:\n"                       <--- THIS
```

Once the current step is completed the `llm_prefix` is added to the next step's prompt. By default, the prefix is `Thought:`, which the llm interprets as "Give me a thought and quit".  By changing the prefix to `New Thought Chain:\n` you entice the model to create a whole new react chain.

**Comment by syker-uk** [comment]
You should probably update issue #1358 with my solution as well. I've been looking at the thread for months (I know) and they are still stuck at retrying rather than fixing the prefix like I did.

**Comment by andrejsstepanovs** [comment]
I was struggling with same issue. This happens mostly only using weaker or LLMs or models with small context. Solution that I found by debugging internals for langchain was to repeat actual format that we expect model to respond with. This works really reliable together with a $ tip than just telling model to 'remember what I told you about answer format'. Sadly I failed to find a good way how one can recover without calling LLM again.

Here is my handle error prompt:
````py
def _handle_error(error: Exception) -> str:
    return """I could not parse your answer. 
It is really important and you will get 1000$ tip if you will use the following answer format:
```
Thought: Do I need to use a tool? Yes
Action: the action to take, should be one of tools mentioned before.
Action Input: the input to the action
Observation: the result of the action
```

When you have a response to say to the Human, or if you do not need to use a tool, you MUST use the format:

```
Thought: Do I need to use a tool? No
AI: [your response here]
```

Please answer again while complying to rules and format just mentioned!
"""
````

and of course pass it in as error handler response:
```py
agent = initialize_agent(
    handle_parsing_errors=_handle_error,
    ...
)
```
